starky/src/proof.rs

```diff
@@ -46,15 +52,57 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize> S
     }
 }
 
+/// Circuit version of [`StarkProof`].
+/// Merkle caps and openings that form the proof of a single STARK.
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub struct StarkProofTarget<const D: usize> {
+    /// `Target` for the Merkle cap if LDEs of trace values.
     pub trace_cap: MerkleCapTarget,
+    /// `Target` for the Merkle cap of LDEs of lookup helper and CTL columns.
     pub auxiliary_polys_cap: Option<MerkleCapTarget>,
+    /// `Target` for the Merkle cap of LDEs of quotient polynomial evaluations.
     pub quotient_polys_cap: MerkleCapTarget,
+    /// `Target`s for the purported values of each polynomial at the challenge point.
     pub openings: StarkOpeningSetTarget<D>,
+    /// `Target`s for the batch FRI argument for all openings.
     pub opening_proof: FriProofTarget<D>,
 }
 
 impl<const D: usize> StarkProofTarget<D> {
+    /// Serializes a STARK proof.
+    pub fn to_buffer(&self, buffer: &mut Vec<u8>) -> IoResult<()> {
+        buffer.write_target_merkle_cap(&self.trace_cap)?;
+        if let Some(poly) = &self.auxiliary_polys_cap {
+            buffer.write_bool(true)?;
+            buffer.write_target_merkle_cap(poly)?;
+        }
+        buffer.write_target_merkle_cap(&self.quotient_polys_cap)?;
+        buffer.write_target_fri_proof(&self.opening_proof)?;
+        self.openings.to_buffer(buffer)?;
+        Ok(())
+    }
+
+    /// Deserializes a STARK proof.
+    pub fn from_buffer(buffer: &mut Buffer) -> IoResult<Self> {
+        let trace_cap = buffer.read_target_merkle_cap()?;
+        let auxiliary_polys_cap = if buffer.read_bool()? {
```

Comment:
I might be totally wrong, but doesn't this read the next value anyway? (so that if there's no bool, we'd read the next value but "discard" it when it should be part of the `quotient_polys_cap`?)

Reply:
I'm not sure to understand this comment. If it refers to the problematic write of boolean you mentioned above then yes. Considering the above fixed, then there is no issue here anymore (we always will have a bool to read, and read the rest only if we do have auxiliary poly cap).